frontend/public/components/catalog-items.jsx

```diff
@@ -169,26 +209,104 @@ export class CatalogTileViewPage extends React.Component {
     return _.find(allCategories, { id: categoryID }).label;
   }
 
+  filterItems() {
+    const { filters } = this.state;
+    const { byName, byType } = filters;
+    const { items } = this.props;
+
+    let filteredItems = [];
+
+    if (byType.clusterServiceClass.active) {
+      filteredItems = _.filter(items, (item) => item.kind === byType.clusterServiceClass.value);
+    }
+
+    if (byType.imageStream.active) {
+      filteredItems = filteredItems.concat(_.filter(items, (item) => item.kind === byType.imageStream.value));
+    }
+
+    filteredItems = !_.isEmpty(filteredItems) ? filteredItems : items;
+
+    if (byName.active) {
+      const filterString = byName.value.toLowerCase();
+      return _.filter(filteredItems, item => item.tileName.toLowerCase().includes(filterString));
+    }
+
+    return filteredItems;
+  }
+
+  clearFilters() {
+    const filters = { ...this.state.filters };
+    filters.byName.active = filters.byType.clusterServiceClass.active = filters.byType.imageStream.active = false;
+    filters.byName.value = '';
+    this.setState({ filters });
+  }
```

Comment:
I realize the filtering is pretty hard-coded, I can adjust to make it more dynamic ie. an array of filter types.